doc/src/guides/solving/solve-equation-algebraically.md

```diff
@@ -0,0 +1,279 @@
+# Solve an equation algebraically
+
+Use SymPy to solve an equation algebraically (symbolically). For example, solving $x^2 = y$ yields $x \in \{-\sqrt{y},\sqrt{y}\}$.
+
+There are two high-level functions to solve equations, {func}`~.solve` and {func}`~.solveset`. Here are recommendations on when to use:
+
+- {func}`~.solve`
+    - You need to programmatically extract components (expressions, or individual symbols or constants) from the output by traversing it.
+    - You want an explicit solution expression that you can use to substitute into something else.
+
+- {func}`~.solveset`
+    - You want a consistent input and output interface.
+    - You want to get all the solutions, including if there are infinitely many.
+    - You want a clear separation between equations in the complex domain and the real domain.
+
+We recommend you include the variable to be solved for as the second argument for either function. While this is optional for equations with a single symbol, it is a good practice because it ensures SymPy will solve for the desired symbol. For example, you may expect the following to solve for $x$, and SymPy will solve for $y$:
+
+```py
+>>> from sympy.abc import x, y
+>>> from sympy import solve
+>>> solve(x ** 2 - y)
+[{y: x**2}]
+```
+
+Specifying the variable to solve for ensures that SymPy solves for it:
+
+```py
+>>> from sympy.abc import x, y
+>>> from sympy import solve
+>>> solve(x ** 2 - y, x)
+[-sqrt(y), sqrt(y)]
+```
+
+## Using {func}`~.solve`
+
+- produces various output formats depending on the answer
+    - unless you use `dict=True` to ensure the result will be formatted as a dictionary, which we recommend if you want to extract information from the result programmatically
```

Comment:
This is related to the discussion that is happening on the mailing list. To me, the second solution is not right. If I tell solve to solve for `[a, b]`, then I don't want `a` or `b` to be included in the solutions. Is this the only difference that occurs when passing a list for a single equation? This is the first time I've seen this advice.

Reply:
> Is this the only difference that occurs when passing a list for a single equation? This is the first time I've seen this advice.

There could be a lot more cases. They go through completely different code paths and `_solve_system` does a lot before it falls back on `_solve`.

Library code usually doesn't use solve with a single expression so the list case is more well tested and fixed.

> To me, the second solution is not right. If I tell solve to solve for `[a, b]`, then I don't want `a` or `b` to be included in the solutions.

This is just how solve represents the solution of an underdetermined system:
```python
In [1]: solve([x + y + z, x - 1], [x, y, z])
Out[1]: {x: 1, y: -z - 1}

In [2]: linsolve([x + y + z, x - 1], [x, y, z])
Out[2]: {(1, -z - 1, z)}

In [3]: nonlinsolve([x + y + z, x - 1], [x, y, z])
Out[3]: {(1, -z - 1, z)}

In [4]: groebner([x + y + z, x - 1], [x, y, z])
Out[4]: GroebnerBasis([x - 1, y + z + 1], x, y, z, domain=ℤ, order=lex)

In [5]: linsolve(_, [x, y])
Out[5]: {(1, -z - 1)}
```
There are other ways but there needs to be a free parameter in the solution so it either needs to be one of the variables or it needs to be a new symbol like the `Ci` that `dsolve` uses to represent families of solutions.

The idea behind `solve` us fundamentally that you want an explicit representation of the solution set so that you can substitute it into to something else to eliminate the symbols given as unknowns. With an underdetermined system it isn't possible to get a substitution rule that will eliminate all unknowns but we can make a substitution rule that will eliminate a subset of the unknowns. The result after substituting is then given in terms of the free parameters which can be a subset of the unknowns or otherwise need to be new symbols. The choice made by `solve` is not to introduce new symbols.

Reply:
Use dict=True in all solve() examples.

Reply:
> Is this the only difference that occurs when passing a list for a single equation? This is the first time I've seen this advice.

See also here where it looks like a bug in heurisch can be fixed by passing a list to solve instead of a single equation:
https://github.com/sympy/sympy/issues/23718#issuecomment-1173733427